collectors/0/druid.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+
+import calendar
+import json
+import re
+import subprocess
+import sys
+import threading
+import time
+
+from dateutil import parser as dtparser
+
+
+DRUID_METRICS_DIR = '/var/log/druid/metrics'
+DRUID_ROLES = [
+        'coordinator',
+        'overlord',
+        'historical',
+        'broker',
+        'realtime'
+        ]
+ 
+EVENT_REGEX = re.compile('Event \[(.*)\]')
+ILLEGAL_CHARS_REGEX = re.compile('[^a-zA-Z0-9\-_./]')
+
+
+def format_tsd_key(metric_key, metric_value, time_, tags={}):
+    def sanitize(s):
+        return ILLEGAL_CHARS_REGEX.sub('_', s)
+
+    expanded_tags = ''.join([' %s=%s'.format(sanitize(key), sanitize(value)) for key, value in tags.iteritems()])
+    output = '{} {} {} {}'.format(sanitize(metric_key), time_, metric_value, expanded_tags)
+    return output
+ 
+
+def report(line):
+    match = EVENT_REGEX.match(line.strip())
+
+    if not match:
+        return
+    
+    event = json.loads(match.group(1))
+
+    metric = 'druid.' + event['metric'].replace('/', '.')
+    value = event['value']
+    timestamp = calendar.timegm(dtparser.parse(event['timestamp']).utctimetuple())
+
+    tags = event.copy()
+
+    del tags['metric']
+    del tags['value']
+    del tags['timestamp']
+
+    print format_tsd_key(metric, value, timestamp, tags)
+
+ 
+def spawn_monitor(metric_file):
+    name = 'monitor for ' + metric_file
+
+    monitor = subprocess.Popen(
+            ['/usr/bin/tail', '--follow=name', '--retry', metric_file],
```

Comment:
If you only want newly appended lines you need to add the option `-n0` otherwise this will first print out the last 10 lines of the file and then print out newly added lines.